core/mondoo-dockerfile-security.mql.yaml

```diff
@@ -0,0 +1,224 @@
+        ```
+        Make sure that `appuser` is created and has the necessary permissions for the processes in that stage.
+  - uid: mondoo-docker-security-use-copy-instead-of-add
+    title: Use COPY Instead of ADD in Dockerfiles
+    impact: 100
+    mql: |
+      docker.file.stages.all(add == empty)
+    docs:
+      desc: |
+        Ensure that Dockerfiles use the `COPY` instruction instead of `ADD`, unless `ADD`'s specific features are needed.
+        The `COPY` instruction is simpler and more predictable, as it only copies files from the source to the destination.
+        `ADD` has additional functionalities such as remote URL fetching and automatic extraction of tar archives, which can introduce security risks if misused.
+        Using `COPY` minimizes these risks by limiting the operations to straightforward file copying.
+      remediation: |
+        Review the Dockerfile and replace `ADD` instructions with `COPY` where possible. Use `ADD` only when its additional functionalities (e.g., fetching files from a remote URL or extracting tar files) are specifically needed and cannot be achieved using `COPY`.
+        Consider the following actions:
+        - Replace `ADD` with `COPY` for file copying tasks.
+        - Use `ADD` only for remote file fetching or unpacking archives if absolutely necessary.
+        - Verify the necessity of each `ADD` instruction and ensure it is used correctly.
+        - Perform a security review to ensure that any use of `ADD` does not introduce vulnerabilities or expose sensitive information.
+  - uid: mondoo-docker-best-practice-no-latest-tag
+    title: Don’t Build Containers from Latest Tags
+    impact: 100
+    mql: |
+      docker.file.stages.all(from.tag != "latest")
+    docs:
+      desc: |
+        Ensure that Dockerfiles do not use the `latest` tag for base images in the `FROM` instructions.
+        Using the `latest` tag can lead to unpredictable builds and potential security issues, as the base image version may change over time.
+      remediation: |
+        Review the Dockerfile to ensure that explicit version tags are used for base images instead of `latest`.
+        For example, use `python:3.9` instead of `python:latest` to ensure consistent and predictable builds.
+  - uid: mondoo-docker-best-practice-use-apt-get
+    title: Prefer `apt-get` Over `apt` for consistent package management
+    impact: 100
+    mql: |
+      docker.file.stages.all(run.none(script.contains("apt")))
+    docs:
+      desc: |
+        Ensure that Dockerfiles use the `apt-get` CLI instead of `apt`. The `apt-get` CLI provides more predictable behavior in scripting contexts and is generally preferred for use in Dockerfiles for its consistency and reliability.
+      remediation: |
+        Review the Dockerfile `RUN` instructions to replace any `apt` commands with `apt-get`. This will ensure that package management operations are performed using the recommended and more stable command-line interface.
```

Comment:
```suggestion
        Review the Dockerfile `RUN` instructions to replace any `apt` commands with `apt-get`. This ensures that package management operations are performed using the recommended and more stable CLI.
```
We already abbreviated it so let's do it again. Our audience knows what CLI stands for. 😄